price/src/price_tracker.rs

```diff
@@ -111,21 +181,18 @@ async fn process_file(
         .await?
         .map_err(Error::from)
         .and_then(|buf| async { PriceReportV1::decode(buf).map_err(Error::from) })
+        .and_then(|report| async move {
+            Price::try_from(&report).map(|price| (report.token_type(), price))
+        })
         .map_err(|err| {
             tracing::warn!("PriceTracker: skipping price report due to error {err:?}");
             err
         })
         .filter_map(|result| async { result.ok() })
-        .for_each(|report| async move {
-            sender.send_if_modified(
-                |prices: &mut Prices| match prices.get(&report.token_type()) {
-                    Some(price) if price == &report.price => false,
-                    _ => {
-                        prices.insert(report.token_type(), report.price);
-                        true
-                    }
-                },
-            );
+        .for_each(|(token_type, price)| async move {
+            sender.send_modify(|prices| {
```

Comment:
so we're always updating now?

Reply:
Yes, because we need to have the latest_timestamp when we try to get the price to verify that it is not too old.

Reply:
make sense